src/app/_lib/contentful.ts

```diff
@@ -0,0 +1,111 @@
+import type {
+  ChainModifiers,
+  Entry,
+  EntryFieldTypes,
+  EntrySkeletonType,
+  LocaleCode,
+} from "contentful";
+import { createClient } from "contentful";
+import { CONTENTFUL_ACCESS_TOKEN, CONTENTFUL_SPACE_ID } from "../_constants";
+import { documentToPlainTextString } from "@contentful/rich-text-plain-text-renderer";
+import { Document } from "@contentful/rich-text-types";
+import words from "lodash.words";
+
+const contentType = "acrossBlogPost";
+const averageReadingSpeed = 238; // words per minute
+
+type TypeAcrossBlogPostFields = {
+  title: EntryFieldTypes.Symbol;
+  slug: EntryFieldTypes.Symbol;
+  content: EntryFieldTypes.RichText;
+  tag: EntryFieldTypes.Array<EntryFieldTypes.Symbol>;
+  featuredImage: EntryFieldTypes.AssetLink;
+  publishDate: EntryFieldTypes.Date;
+  description: EntryFieldTypes.Symbol;
+};
+
+type TypeAcrossBlogPostSkeleton = EntrySkeletonType<
+  TypeAcrossBlogPostFields,
+  "acrossBlogPost"
+>;
+export type BlogPostType = Entry<
+  TypeAcrossBlogPostSkeleton,
+  "WITHOUT_UNRESOLVABLE_LINKS",
+  string
+>;
+
+function getProductionClient() {
+  return createClient({
+    space: CONTENTFUL_SPACE_ID ?? "",
+    accessToken: CONTENTFUL_ACCESS_TOKEN ?? "",
+  });
+}
+
+export async function retrieveContentfulPublishedSlugs(): Promise<string[]> {
+  const client = getProductionClient();
+  const options = {
+    content_type: contentType,
+    select: "fields.slug",
+    "fields.content[exists]": true,
+    "fields.slug[exists]": true,
+  } as const;
+  const entries =
+    await client.withoutUnresolvableLinks.getEntries<TypeAcrossBlogPostSkeleton>(options);
+  return entries.items.map((item) => item.fields.slug);
+}
+
+export async function retrieveContentfulEntry(
+  entrySlugId: string,
+  relevantEntryCount = 4,
+) {
+  const client = getProductionClient();
+  const options = {
+    content_type: contentType,
+    limit: 1,
+    "fields.slug[match]": entrySlugId,
+  } as const;
+  const entries =
+    await client.withoutUnresolvableLinks.getEntries<TypeAcrossBlogPostSkeleton>(options);
+  const entry = entries.items[0];
+  if (!entry) {
+    return undefined;
+  }
+  const relevantEntries = await retrieveRelevantContentfulEntries(
+    entrySlugId,
+    entry.fields.tag ?? [],
+    relevantEntryCount,
+  );
+  return {
+    ...entry,
+    relevantEntries,
+  };
+}
+
+export async function retrieveRelevantContentfulEntries(
+  entrySlugId: string,
+  tags: string[],
+  limit: number,
+): Promise<BlogPostType[]> {
+  const client = getProductionClient();
+  const options = {
+    content_type: contentType,
+    limit,
+    "fields.content[exists]": true, // no empty posts
+    "fields.tag[in]": tags.join(","), // get posts with same tags
```

Comment:
be careful here: this is **case sensitive**.  So, for example, the string `uma` will not match any entries if the tag is `UMA` on contentful.

Reply:
Great callout +1

Reply:
https://github.com/across-protocol/across.to/pull/56/commits/0cdc9664dfd3f35884e5f3890b76030b351c5727